cmd/zed/zed.d/statechange-led.sh

```diff
@@ -94,15 +99,27 @@ check_and_set_led()
 
 state_to_val()
 {
-	state="$1"
-	case "$state" in
-		FAULTED|DEGRADED|UNAVAIL)
+	led="$1"
+	state="$2"
+
+	if [ "$led" = "fault" ]; then
+		case "$state" in
+			FAULTED|DEGRADED|UNAVAIL)
+				echo 1
+				;;
+			ONLINE)
+				echo 0
+				;;
+		esac
+	elif [ "$led" = "locate" ]; then
+		if [ "$state" = "OFFLINE" ]; then
 			echo 1
-			;;
-		ONLINE)
+		else
 			echo 0
-			;;
-	esac
+		fi
+	else
+		zed_log_msg "$0 state_to_val error: led=$led not understood"
+	fi
```

Comment:
This might be better served as something like
```sh
case "$led" in
	fault)
		case "$state" in
			FAULTED|DEGRADED|UNAVAIL) echo 1 ;;
			ONLINE)                   echo 0 ;;
		esac
		;;
	locate)
		case "$state" in
			OFFLINE) echo 1 ;;
			*)       echo 0 ;;
		esac
		;;
	*)
		zed_log_msg "$0 state_to_val error: led=$led not understood"
		;;
esac
```

Reply:
In a271946d724ae1d399564ace759fce8e00686812 - but superseeded by 138c8d24066bc228ccfa31a3d7a1d30629bdf2e8